src/services/slack.ts

```
@@ -46,7 +44,8 @@ export const setUserPresence = async (email: string, token: string | undefined,
   const slackClient = new WebClient(token);
 
   try {
-    await slackClient.users.setPresence({ presence });
+    const result = await slackClient.users.setPresence({ presence });
+    throw result.error;
```

Do we want to throw the error or just log it?

---

I think throwing is still good 🤔 The handleError function ends up checking the error message and if it's "token_revoked" or "invalid_auth"  it clears the tokens and sends the user a message in Slack to re-auth.

---

Does this need to be wrapped in a conditional statement then to only throw if there is an error?

---

Swapped to the the response `ok` boolean and only throw if that's missing/false!
[2079346](https://github.com/hudl/CalendarToSlack/pull/96/commits/2079346de455257bba346d6c58f23420e848e340)